gadopt/utility.py

```diff
@@ -475,3 +477,56 @@ def node_coordinates(function):
     return [
         Function(func_space).interpolate(coords).dat.data for coords in mesh_coords
     ]
+
+
+def assign_1d_profile(function: Function, one_d_filename: str, cartesian: bool, r1d=None):
+    """
+    Assign a one-dimensional profile to a Function `q` from a file.
+
+    The function reads a one-dimensional radial viscosity profile together with the
+    radius input for the profile, from a file, broadcasts the read array to all
+    processes, and then interpolates this array onto the function space of `q`.
+
+    Args:
+        q (Function): The function onto which the 1D profile will be assigned.
+        one_d_filename (str): The path to the file containing the 1D radial viscosity profile.
+        cartesian (bool): Is the upward direction along z/y(cartesian=True) or radial (cartesian=False)
+
+    Returns:
+        None: It directly assigns the input function `function`.
+
+    Note:
+        - Note the cartesian flag
+        - This is designed to read a file with one process and distribute in parallel with MPI.
+        - The input file should contain an array of radius and an array of viscosity values, seperated by comma.
+        - Make sure radius in the file is consistent with the mesh.
+    """
+    # find the mesh
+    mesh = extract_unique_domain(function)
+
+    visc = None
+    rshl = None
+
+    # read the input file
+    if mesh.comm.rank == 0:
+        # The root process reads the file
+        rshl, visc = np.loadtxt(one_d_filename, unpack=True, delimiter=",")
```

Comment:
The two assignments at the top usually come as part of an `else` block. It adds to readability but is not mandatory, so it is up to you.